src/oplsNipals.js

```diff
@@ -47,32 +44,27 @@ export function oplsNipals(data, labels, options = {}) {
   let t, c, w, uNew;
   for (let i = 0; i < numberOSC && diff > limit; i++) {
     w = u.transpose().mmul(data).div(u.transpose().mmul(u).get(0, 0));
-    w = w.transpose().div(norm(w));
-
+    w = w.transpose().div(w.norm());
     t = data.mmul(w).div(w.transpose().mmul(w).get(0, 0)); // t_h paso 3
 
     // calc loading
     c = t.transpose().mmul(labels).div(t.transpose().mmul(t).get(0, 0));
 
     // calc new u and compare with one in previus iteration (stop criterion)
     uNew = labels.mmul(c.transpose());
-    uNew = uNew.div(c.transpose().mmul(c).get(0, 0));
-
+    uNew = uNew.div(c.norm() ** 2);
```

Comment:
where does this multiplication by 2 comes from?  I think c.transpose().mmul(c) is indeed the norm. This was wrong because we only got the first value. So this correction makes sense, but I cannot understand why multiplying by 2

Reply:
No, in this case `c` is a vector  and `c·c = c.transpose().mmul(c)` is the scalar product, that's the same as norm of C to the power of 2.